lib/internal/streams/end-of-stream.js

```diff
@@ -92,7 +97,23 @@ function eos(stream, opts, callback) {
   if (opts.error !== false) stream.on('error', onerror);
   stream.on('close', onclose);
 
+  const closed = (w && w.closed) || (r && r.closed) ||
+    (w && w.errorEmitted) || (r && r.errorEmitted) ||
+    (w && w.finished) || (r && r.endEmitted) ||
+    (r && stream.req && stream.aborted);
+
+  if (closed) {
+    // TODO(ronag): Re-throw error?
```

Comment:
Nit: a bit more context for these Todo comments would be good. If someone else comes in here and sees these there's not going to be enough information here for them to know what the considerations are.